style.md

```diff
@@ -151,6 +152,65 @@ An interface is two fields:
 If you want interface methods to modify the underlying data, you must use a
 pointer.
 
+### Verify Interface Compliance
+
+For types that are known or required to implement specific interfaces, add
+checks to verify interface compliance at compile time.
```

Comment:
Should this still be done if the constructor for the type has a signature that returns the interface, but the implementation returns the struct? 

Does it also mean that every type that implements `Stringer` should do this?

Reply:
Per https://github.com/uber-go/guide/issues/25#issuecomment-541377764, I figured we were going to add the recommendation without covering things like "it's not necessary if a constructor or another function returns it."

RE: stringer: Switched away from it. It's usually unnecessary to verify Stringer compliance but sometimes it's part of the contract, in which case you do want it.

I previously used phrasing that asks this only for "types that are known or required to implement specific interfaces." We can narrow it down to "types that are required to implement specific interfaces," making this a requirement only if it's part of the contract of a type.

Reply:
Oh and if we add the 'exported' qualifier, it covers the case of constructors returning unexported implementations.